acpi: add dbs-acpi support for Dragonball

studychao opened the pull request:
This is for initializing dbs-acpi.

fixes: #253

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
boot: prepare to publish v0.3.1

- Update kvm-bindings to v0.6.0
- Use caret dependencies for kvm-ioctls

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
interrupt: prepare to publish 0.2.2

Update kvm-bindings to 0.6.0
Update kvm-ioctls to 0.12.0

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
virtio-queue: update virtio-queue to v0.6.0

Since the vm-memory conflict brought by virtio-queue 0.4.0, we need to
upgrade virtio-queue to the newest version. Meanwhile, virtio-queue
change a lot of APIs from 0.4.0 to 0.5.0, so this PR mainly for fixing
all the API changes.

fixes: #249
Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

xuejun-xj committed:
dbs-arch: update kvm-ioctls to 0.12.0

Update kvm-ioctls for:
1. support 128-bit SET and GET registers.
2. support SET and GET attr ioctl interfaces for vcpu fd.

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

xuejun-xj committed:
dbs-arch: support PMU virtualization on aarch64

Add pmu.rs to support PMu virtualization on aarch64. It exposes
three interfaces to help check and initialize pmu feature on
vcpu devices.

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

xuejun-xj committed:
dbs-boot: update fdt for aarch64

This commit contains these changes:
1. Add create_pmu_node function to write pmu into fdt.
2. Refector fdt UT implementation.
3. Add UT for pmu node, test_create_fdt_with_pmu().
4. Tidy up use dependencies in mod tests.
5. Update dtb files.

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

HerlinCoder committed:
dbs-virtio-devices: introduce virtio-mem device

Introduce virtio-mem device, to support dynamic resizing of virtual
machine memory.

Signed-off-by: Hui Zhu <teawater@antgroup.com>
Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

HerlinCoder committed:
dbs-virtio-devices: introduce virtio-balloon device

Introduce virtio-balloon device, to support dynamic resizing of virtual
machine memory.

Signed-off-by: Hui Zhu <teawater@antgroup.com>
Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

studychao committed:
nydus: update nydus api, blobfs and rafs

In order to solve virtio-queue dependency problem, we need to update
nydus crates.

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
crates: update ut related to virtio-queue

Update the unit tests that are related to virtio-queue upgrade changes.

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
Clippy: fix clippy error

fix several clippy errors introduced by the latest Rust.

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
ut: fix ut errors due to max queue size

adjust some queue size in UT to fix queue size erorr.

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
acpi: add dbs-acpi support for Dragonball

This is for initializing dbs-acpi.

fixes: #253

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

wllenyj committed:
ci: use clippy directly instead of clippy-check

Signed-off-by: wllenyj <wllenyj@linux.alibaba.com>

studychao committed:
nydus: update nydus related crates

update nydus-api, nydus-blobfs and nydus-rafs to fit with the
virtio-queue version.

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
clippy: fix new clippy warnings introduced by rust 1.67

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

xuejun-xj committed:
dbs-upcall: add vcpu hotplug support for aarch64

Add vcpu resize support for upcall on aarch64 and update related UT.

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

xuejun-xj committed:
dbs-interrrupt: fix cargo test on aarch64

Create gicv3 device as interrupt controller instead of irqchip.

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

studychao committed:
upcall: change reconnection log to use debug!

In order to reduce the similar log, we change reconnection log to
debug!.

fixes:#251
Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

HerlinCoder committed:
dbs-address-space: add prot_flags to AddressSpaceRegion

We add prot_flags to AddressSpaceRegion so that we can access this field
when we needed for a corresponding guest address or a region.

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

HerlinCoder committed:
dbs-address-space: add USABLE_END to get host memory size

We add a new reference USABLE_END to get total memory size on host,
which can be use for setting capacity of virtio-mem.

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

studychao committed:
arch: update to version 0.2.2

- Update kvm-bindings to v0.6.0
- Update kvm-ioctls to v0.12.0
- Use caret dependencies for kvm-ioctls
- vPMU support in aarch64

fixes: #257
Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
virtio: release virtio device 0.2.0

- Update kvm-ioctls to v0.12.0
- Update kvm-bindings to v0.6.0
- Update virtio-queue to v0.6.0
- Update Nydus API to adjust to the latest Nydus version

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

wllenyj commented on crates/dbs-acpi/src/rsdp.rs:
```diff
@@ -0,0 +1,59 @@
+//
+// SPDX-License-Identifier: Apache-2.0
+// RSDP (Root System Description Pointer) is a data structure used in the ACPI programming interface.
+use vm_memory::ByteValued;
```
Add a new blank line?

studychao committed:
upcall: update virtio devices to 0.2.0

- update dbs-virtio-devices to v0.2.0

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

HerlinCoder committed:
dbs-address-space: update unit test

We rename some unit test to make it more readable.

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

xuejun-xj committed:
bugfix: dbs-arch: read_mpidr return u64

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

HerlinCoder committed:
virtio-devices: balloon get pfn as u32

pfn should be parsed as u32 instead of u64 in virtio-balloon.

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

HerlinCoder committed:
virtio-device: update virtio devices to v0.3.0

update virtio devices to v0.3.0

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

HerlinCoder committed:
dbs-upcall: update to v0.3.0

update dbs-virtio-devices to v0.3.0

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

HerlinCoder committed:
dbs-address-space: update to v0.3.0

- add prot_flags to AddressSpaceRegion
- add USABLE_END to get host memory size

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

studychao committed:
dbs-arch : release v0.2.3

Fixed:
- read_mpidr return u64 instead of u128

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
doc: add crate links in README and fix crate-release.md

1. add crate links to crates.io in project README
2. fix wrong steps in crate-release.md

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

xuejun-xj committed:
dbs-boot: fdt: add a switch of creating dtb files

Add a switch of creating dtb files for fdt unit tests. It will read
environment variable "FDT_CREATE_DTB" to decide whether to create new
dtb file for this "cargo test".

Usage:
```shell
    $ FDT_CREATE_DTB=1 cargo test
```

Fixes: #276
Opensource: y

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

xuejun-xj committed:
dbs-boot: fdt_utils: create a new module

This commit creates a new module called fdt_utils. It abstracts the
parameters for constructing fdt on aarch64. It contains three structs
for now:
1. FdtVmInfo: saves information about micro vm
2. FdtNumaInfo: saves information about numa
3. FdtDeviceInfo: saves information about devices

Fixes: #276
Opensource: y

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

justxuewei committed:
dbs-virtio-devices: Clear resources when removing virtio devices

This PR implements functionaility of clearing resources for virtio-net
and virtio-vsock devices.

For virtio-net devices, the removed resource is its event hanlder.

For virtio-vsock devices, the removed resources are its event handler
and a socket on the host.

Fixes: #273

Signed-off-by: Zizheng Bian <zizheng.bian@linux.alibaba.com>
Signed-off-by: Xuewei Niu <niuxuewei.nxw@antgroup.com>

justxuewei committed:
dbs-utils: Fix clippy issues caused by rust 1.68.2

Obtaining reference of unaligned fields is being phased out. Therefore,
`&struct.field` is replaced with `std::ptr::addr_of!(struct.field)` to
avoid those warning issues. For more details, please refer to
https://github.com/rust-lang/rust/issues/82523.

Fixes: #273

Signed-off-by: Xuewei Niu <niuxuewei.nxw@antgroup.com>

xuejun-xj committed:
dbs-boot: fdt: redefines vcpu_mpdir to vcpu_state

This commit redefines the parameter "vcpu_mpidr" to "vcpu_state".
"vcpu_state" combines two cpu attributes: mpidr and boot_onlined. The
"boot_onlined" attribute indicates whether this cpu should be brought up
at boot procedure. 1 means online while 0 means offline.

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

xuejun-xj committed:
dbs-boot: release v0.4.0

Updated:
- update create_fdt interface
- simplify create_fdt parameters
- refector fdt UT

Signed-off-by: xuejun-xj <jiyunxue@linux.alibaba.com>

HerlinCoder committed:
dbs-virtio-devices: add Madvise error type

We add Madvise error type to expose error when setting madvise on guest
memory region.

Fixes: #285

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

HerlinCoder committed:
dbs-virtio-devices: fix balloon pfn using u32

We fixes a problem that offset should be parsed as u32 instead of
u64 in virtio-balloon.

Fixes: #284

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

HerlinCoder committed:
virtio-device: update to v0.3.1

Fixed:
- Fixed #284, fix balloon pfn using u32
- Fixed #285, add Madvise error type

Signed-off-by: Helin Guo <helinguo@linux.alibaba.com>

lisongqian committed:
Forward correct evset in vsock muxer

When forwarding an epoll event from the unix muxer to the
targeted connection event handler, the eventset the connection
registered is forwarded instead of the actual epoll
operation (IN/OUT).

For example, if the connection was registered for EPOLLIN,
and receives an EPOLLOUT, the connection will actually handle
an EPOLLOUT.

This is the root cause of previous bug, which caused the
introduction of some workarounds (i.e: handling ewouldblock
when reading after receiving EPOLLIN, which should never happen).

When matching the connection, we retrieve and use the evset of
the connection instead of the one passed as a parameter.
The compiler does not complain for an unused variable because
it was first logged in a debug! statement.

This is an unfortunate naming mistake that caused a lot of problems.

Signed-off-by: lisongqian <mail@lisongqian.cn>

studychao committed:
dbs-virtio-devices: update Nydus to fit with the upstream Nydus updates

Since Nydus image service was updated, and that breaks our compilation
of dragonball-sandbox.

In order to fix that, we need to update the nydus dependencies in
dragonball-sandbox.

Also, some clippy warnings & build warnings introduced by Rust 1.70 are
fixed.

fixes: #293

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
rust: update to Rust 1.70

fix several build and clippy warnings for Rust 1.70.
Also, epoll update to 4.3.2 which bitflag is also updated to version 2.
In this way, bitflag version 2 is conflict with other bitflag used under
version 2. So we fix epoll in 4.3.1 in this commit.

fixes: #293

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

studychao committed:
doc: update the notice for dragonball-sandbox future

a bright new start.

Signed-off-by: Chao Wu <chaowu@linux.alibaba.com>

NK-fengsf commented:
looks good for me

studychao commented:
Close this and #297 will continue the work.
